### PR TITLE
Clarifies events page

### LINKED ISSUE #5393: [FEATURE] Adding clarity to the Events page
### Description

1. Currently, the event page has 6 filter buttons, which almost take up the whole screen. It would be great to decrease the size and align all the buttons side by side. Or else, it would be great to have a drop-down.

2.  Currently, I guess the default filter is "All Future Events". It would be great if we can make the default filter "All Events". Then the user can filter it as per their needs.

### Screenshots

image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by akash19coder**
## Fixes Issue  

closes #5393 

## Changes proposed
 Added clarity to the events page by introducing a dropdown.
Create a DropDown component.

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7180"></a>

### CODE REVIEW

**Review by amandamartin-dev — DISMISSED** [review]
A few notes.  Would it be possible to only have the dropdown show valid choices?  It's not a great UX to click through many that have "no events found".

Another thing I noticed was css related.  Some of the text in the dropdown overlaps the carrot (I'll add some screenshots).  There could also be some more space between the colon and dropdown.

And lastly just a point for discussion - this looks a little unusual to me on mobile although I'm not sure a better solution off the top of my head. Would love another set of eyes here wdyt @eddiejaoude 

mobile view:
mobile

style overlap/spacing:
styling issues

**Comment by akash19coder** [comment]
> A few notes. Would it be possible to only have the dropdown show valid choices? It's not a great UX to click through many that have "no events found".

I have changed the dropdown to show only the valid choices.

> Another thing I noticed was css related. Some of the text in the dropdown overlaps the carrot (I'll add some screenshots). There could also be some more space between the colon and dropdown.

Fixed this also. Added more space between colon and dropdown.

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍  thank you

**Comment by eddiejaoude** [comment]
> And lastly just a point for discussion - this looks a little unusual to me on mobile although I'm not sure a better solution off the top of my head. Would love another set of eyes here wdyt @eddiejaoude

Thanks @amandamartin-dev for reviewing! Yep the double dropdown doesn't look quite right, but I am not sure of another solution - I don't think it is a block for now, we can get it out and people can feedback with ideas (we are also looking into getting a redesign of the app at the moment so it could get moped up with that)

**Comment by eddiejaoude** [comment]
Changes look good...


Screenshot 2023-05-30 at 10 30 22
Screenshot 2023-05-30 at 10 30 10

**Comment by eddiejaoude** [comment]
@akash19coder please can you resolve the conflict. I think it acme from this PR https://github.com/EddieHubCommunity/LinkFree/pull/7135

**Comment by akash19coder** [comment]
@eddiejaoude I have resolved all the conflicts
😄

File: components/user/UserEvents.js [hunk 1]
```diff
@@ -1,38 +1,92 @@
...
+    { value: 'free', name: 'Free Events'},
+    { value: 'paid', name: 'Paid Events'},
+    { value: 'past', name: 'Past Events'}
+  ];
+
+  const handleEventTypeChange = (event) => {
+    setEventType(event.target.value);
+  };
+
+  const filterByEventType = (event, eventType) => {
```
Comment (dan-mba) [inline components/user/UserEvents.js:25]: Doing Date math inside a server-side rendered component is a bad idea.

Please change this section to look like:
```js
  const filterByEventType = (event, eventType) => {
    switch (eventType) {
      case "future":
        return event.date.future;
      case "ongoing":
        return event.date.ongoing;
      case "virtual":
        return event.date.future && event.isVirtual;
      case "inPerson":
        return event.date.future && event.isInPerson;
      case "cfpOpen":
        return event.date.cfpOpen;
      case "free":
        return event.price?.startingFrom === 0;
      case "paid":
        return event.price?.startingFrom > 0;
      case "past":
        return !event.date.future;
      default:
        return true;
    }
  };
```

**Review by dan-mba — CHANGES_REQUESTED** [review]

File: components/user/UserEvents.js (hunk 1 above)
Comment (akash19coder) [reply components/user/UserEvents.js:25]: @dan-mba I have changed the function to lool like what you suggested

**Review by dan-mba — APPROVED** [review]
Looks good to me

**Pull request merged**